### PR TITLE
fix(TomlTable): fixes key serialization by adding better key name checks

### LINKED ISSUE #30: Keys with symbols aren't quoted but they should be
I have a dictionary like so:
```c#
var obj = new Dictionary<string, string>()
{
    {"normal-key", "value for normal key"},
    {"key with spaces", "value for key with spaces"},
    {"key!with{}(*%&)random[other+symbols", "value for key with random other symbols"},
    {"key/with/slashes", "value for key with slashes"},
};
```
And I'm saving it with `TomletMain.TomlStringFrom(obj);`, but some of the keys in the output are not quoted even though they should be:
```toml
normal-key = "value for normal key" # correctly not quoted
"key with spaces" = "value for key with spaces" # correctly quoted
key!with{}(*%&)random[other+symbols = "value for key with random other symbols" # should be quoted but isn't
key/with/slashes = "value for key with slashes" # should be quoted but isn't
```

Now this isn't actually causing a problem for me currently since the only non-alphanumeric characters in my keys are forward slashes (which Tomlet can read back just fine), but technically this is against the toml spec and would cause issues with other characters.

Also, I get a `Tomlet.Exceptions.TomlNoSuchValueException` if I try having dictionary keys with full stops in them, but I don't know if that's a problem with this library or if that's against the toml spec.

### PR DESCRIPTION
**Pull request by tcortega**
Uses a foreach to iterate through the key strings both in the table and during object serialization to identify which keys needs to be quoted in a performative way.

~Also made a small optimization in the methods inside `TomlUtils`.~

The tests might not look that good, feel free to change it if you want. Or give me a heads up on what you have in mind for the tests.

Closes #30 
Closes #31

### CODE REVIEW

**Comment by SamboyCoding** [comment]
Can you rebase this onto latest?

File: Tomlet.Tests/TableTests.cs [hunk 1]
```diff
@@ -46,6 +46,26 @@ public void TablesCanHaveQuotedKeyNames()
...
+         [InlineData("normal-key", "normal-key")]
+         [InlineData("normal_key", "normal_key")]
+         [InlineData("normalkey", "normalkey")]
+         [InlineData("\"key with spaces\"", "\"key with spaces\"")]
+         [InlineData("key!with{}(*%&)random[other+symbols", "\"key!with{}(*%&)random[other+symbols\"")]
+         [InlineData("key/with/slashes", "\"key/with/slashes\"")]
+         [InlineData("Nam\\e", "\"Nam\\\\e\"")]
+         public void KeysShouldBeSerializedCorrectly(string inputKey, string expectedKey)
+         {
+             var inputString = $"""
```
Comment (SamboyCoding) [inline Tomlet.Tests/TableTests.cs:60]: Raw string literals aren't supported on the current language version/net6

**Review by SamboyCoding — CHANGES_REQUESTED** [review]

File: Tomlet.Tests/TableTests.cs (hunk 1 above)
Comment (tcortega) [reply Tomlet.Tests/TableTests.cs:60]: Can we move the tests project to .NET 7?

Comment (SamboyCoding) [reply Tomlet.Tests/TableTests.cs:60]: I suppose that's an option yeah. Let me get on that

Comment (SamboyCoding) [reply Tomlet.Tests/TableTests.cs:60]: Alright, if you rebase onto master now it should work

**Review by SamboyCoding — APPROVED** [review]

**Pull request merged**